app/Services/QuizAttemptService.php

```diff
@@ -25,36 +35,44 @@ public function startQuiz($quizId, $data)
         return $quizAttempt;
     }
 
-    public function submitAnswers($quizAttemptId, $data)
-{
-    $quizAttempt = QuizAttempt::query()->findOrFail($quizAttemptId);
-    $score = 0;
-
-    DB::transaction(function () use ($quizAttempt, $data, &$score) {
-        foreach ($data['answers'] as $item) {
-            $questionId = $item['question_id'];
-            $answerId = $item['answer_id'];
-
-            $question = Question::query()->findOrFail($questionId);
-            $answer = Answer::findOrFail($answerId);
-
-            QuizAttemptAnswer::create([
-                'quiz_attempt_id' => $quizAttempt->id,
-                'question_id' => $questionId,
-                'answer_id' => $answerId
-            ]);
-
-            if ($answer->is_correct) {
-                $score++;
-            }
+
+   public function submitAnswers($quizAttemptId, $data)
```

Comment:
this would be a better approach since we fetch the answers outside the transaction and do the insertation and updates inside the transaction, analyze this code and test it, ad ask if you don't understand something